gnd/src/main/java/com/google/android/gnd/workers/FileDownloadWorker.java

```diff
@@ -183,24 +150,21 @@ private Result checkDownload(Tile tile) {
   @NonNull
   @Override
   public Result doWork() {
-    Log.d(TAG, "Downloading tile: " + Tile.pathFromId(tileId));
     Tile tile = localDataStore.getTile(tileId).blockingGet();
 
     // When there is no tile in the db, the maybe completes and returns null.
+    // We expect tiles to be added to the DB prior to downloading.
+    // If that isn't the case, we fail.
     if (tile == null) {
```

Comment:
What if the tile has been deleted since this job was queued? Maybe we shouldn't fail but just exit?

Reply:
Hmm, it seems like the right solution here is to use the workManager to cancel the jobs as soon as the tile is deleted: https://developer.android.com/topic/libraries/architecture/workmanager/how-to/cancel-stop-work. Probably some listener on the DB that fires off `cancelAllWork` when a user deletes an area and all its corresponding tiles are up for deletion. i.e. whatever method we implement that deletes tiles first checks in w/ the manager to see if any work is queued and stops all the relevant work first. -- we'll probably have to tag jobs by area id to ensure we stop work for just that area (and not e.g. other areas that were queued up immediately after)

I think this would appropriately handle that case, while this check explicitly uncovers whether or not we queued this without adding something to the DB first.

Reply:
I don't feel smart enough to consider all the interactions between the workers and managers, so as a defensive maneuver would normally try to make everything idempotent that can be. Besides the user deleting the tile, it could also get remove if the user clears the app cache; I'm not sure if workers get cancelled as well when that happens. Even if they do, it would be nice not to need to know whether that's the case. Perhaps we can just bail out and log a debug log message instead?

Reply:
That sounds fair to me--so, I suppose we need to move this nullness check into the manager and avoid scheduling any work in the first place if we can't find the tile in the DB.

Reply:
We will need the nullness check here regardless, since this logic is executed in a different scheduler than the thread that queues the tile. Currently the manager will only queue work after adding the tiles anyway iirc?